src/playercards/AllCardsBag.ttslua

```diff
@@ -131,70 +254,179 @@ function buildSupplementalIndexes()
         end
       end
 
-      -- Add the card to the appropriate class and level indexes
-      local isGuardian = false
-      local isSeeker = false
-      local isMystic = false
-      local isRogue = false
-      local isSurvivor = false
-      local isNeutral = false
-      local upgradeKey
-      -- Excludes signature cards (which have no class or level) and alternate
-      -- ID entries
-      if (cardMetadata.class ~= nil and cardMetadata.level ~= nil) then
-        isGuardian = string.match(cardMetadata.class, "Guardian")
-        isSeeker = string.match(cardMetadata.class, "Seeker")
-        isMystic = string.match(cardMetadata.class, "Mystic")
-        isRogue = string.match(cardMetadata.class, "Rogue")
-        isSurvivor = string.match(cardMetadata.class, "Survivor")
-        isNeutral = string.match(cardMetadata.class, "Neutral")
-        if (cardMetadata.level > 0) then
-          upgradeKey = "-upgrade"
-        else
-          upgradeKey = "-level0"
+      table.insert(allCardsIndex, cardMetadata.id)
+
+      if cardMetadata.level then
+        table.insert(levelIndex[cardMetadata.level], cardMetadata.id)
+      else
+        otherIndex[cardMetadata.id] = true
+      end
+
+      if cardMetadata.class then
+        if (string.match(cardMetadata.class, "Guardian")) then
+          table.insert(classIndex["guardian"], cardMetadata.id)
+        end
+        if (string.match(cardMetadata.class, "Seeker")) then
+          table.insert(classIndex["seeker"], cardMetadata.id)
         end
-        if (isGuardian) then
-          table.insert(classAndLevelIndex["Guardian"..upgradeKey], cardMetadata.id)
+        if (string.match(cardMetadata.class, "Mystic")) then
+          table.insert(classIndex["mystic"], cardMetadata.id)
         end
-        if (isSeeker) then
-          table.insert(classAndLevelIndex["Seeker"..upgradeKey], cardMetadata.id)
+        if (string.match(cardMetadata.class, "Rogue")) then
+          table.insert(classIndex["rogue"], cardMetadata.id)
         end
-        if (isMystic) then
-          table.insert(classAndLevelIndex["Mystic"..upgradeKey], cardMetadata.id)
+        if (string.match(cardMetadata.class, "Survivor")) then
+          table.insert(classIndex["survivor"], cardMetadata.id)
         end
-        if (isRogue) then
-          table.insert(classAndLevelIndex["Rogue"..upgradeKey], cardMetadata.id)
+        if (string.match(cardMetadata.class, "Neutral")) then
+          table.insert(classIndex["neutral"], cardMetadata.id)
+        end
+      end    
```

Comment:
This should be simplified to just using the values instead of 6 if-clauses. E.g. use the `|` as separator to split the string into X substrings, loop through them and insert the card id into each respective classindex